forecasting/src/autogluon/forecasting/trainer/abstract_trainer.py

```diff
@@ -132,8 +132,11 @@ def save(self) -> None:
         try:
             save_pkl.save(path=self.path_pkl, object=self)
         except:  # noqa
+            for model in self.models.values():
+                model.save()
```

Comment:
This feels spooky. I feel like we should just crash if saving the pkl is raising an exception.

What do you think about removing the `self.low_memory=False` logic entirely? I have stopped maintaining `self.low_memory=False` in Tabular for awhile since most users wouldn't use it and it is very clunky to deal with as many models may internally want to cache to disk during fit. This could simplify our code and minimize defects in future.

Reply:
I completely agree re `low_memory`. I'll revise.

Reply:
Just FYI I left the logic as is in the `SimpleAbstractTrainer` class which we intend to move later to core. We can revise the logic there when we'd like. Left a todo.